### PR TITLE
deCamelCase MOS return values

### PR DESCRIPTION
**Pull request by romecode**
Standardize to EOS. Add test case for translator.

Trigger checks.

### CODE REVIEW

**Review by dcasnowdon — CHANGES_REQUESTED** [review]
This looks good to me. I think the interface to the translator class could be more natural -- i.e. it's not just a method that we should assign, but the class itself, with several methods for different translation activities -- e.g. translating the CLI commands, and translating the returned json data. 

I think we should change that interface to the more natural style just because third parties may wish to build other translators and this is effectively a persistent API.

**Review by romecode — COMMENTED** [review]
Split up the translation functions.

File: src/pytest_netdut/wrappers.py [hunk 1]
```diff
@@ -34,36 +34,86 @@ def _splitcmds(cmds):
...
+
+        Args:
+            data (str | dict): Results are typically a string or a dictionary.
+        """
+        if isinstance(data, dict):
+            return self.process_dict(data)
+        return data
+
+
+class EOS_MOS_Translator(Translator):
```
Comment (alexw-anet) [inline src/pytest_netdut/wrappers.py:96]: Can you give this a PascalCase style name please?

File: src/pytest_netdut/wrappers.py [hunk 2]
```diff
@@ -34,36 +34,86 @@ def _splitcmds(cmds):
...
+    """Standardize config and results to EOS."""
+
+    def __init__(self):
```
Comment (alexw-anet) [inline src/pytest_netdut/wrappers.py:99]: I think this should have a compatible signature with the base class.

Do we want `config_patterns` to be an argument, or is it always a static feature of the class?

File: src/pytest_netdut/wrappers.py [hunk 3]
```diff
@@ -34,36 +34,84 @@ def _splitcmds(cmds):
...
+    Override the standardize_key function to implement result processing.
+
+    Args:
+        config_patterns (iterable[regex pattern]): List of regex patterns for CLI translations.
+    """
+
+    config_patterns = []
+
+    def standardize_key(self, k):
+        """Returns a standardized key.
```
Comment (dcasnowdon) [inline src/pytest_netdut/wrappers.py:50]: It's not clear from this docstring what this key is for. I think this is for translating json results?

File: src/pytest_netdut/wrappers.py [hunk 4]
```diff
@@ -34,36 +34,84 @@ def _splitcmds(cmds):
...
+        return commands
+
+    def json(self, data):
+        """JSON translator.
+
+        Args:
+            data (str | dict): Results are typically a string or a dictionary.
+        """
+        if isinstance(data, dict):
+            return self.process_dict(data)
```
Comment (dcasnowdon) [inline src/pytest_netdut/wrappers.py:91]: Should this be called "translate_json_dict" or similar to make it more clear what this is doing?

**Review by dcasnowdon — APPROVED** [review]
I've added a couple of comments which might be good to fix up -- e.g. the docstrings. 

It would also be good to add some documentation about the translator to the README.md and the docstrings.

**Pull request merged**